module/applications/actor/base-sheet.mjs

```diff
@@ -1159,6 +1134,38 @@ export default class ActorSheet5e extends ActorSheet {
 
   /* -------------------------------------------- */
 
+  /**
+   * Adjust the preparation mode of a dropped spell depending on the drop location on the sheet.
+   * @param {object} itemData    The item data requested for creation. **Will be mutated.**
+   * @param {DragEvent} event    The original drag event.
+   */
+  _onDropSpell(itemData, event) {
+    if ( itemData.type !== "spell" || itemData.system.level == 0 ) return;
+
+    // Determine the section it is dropped on, if any.
+    const mode = event.target.closest("div[data-type='spell']")?.dataset;
+
+    // If the spell is dropped on a section, set the mode to be that type.
+    if ( mode && mode.level > 0 ) itemData.system.preparation.mode = mode["preparation.mode"];
+
```

Comment:
```suggestion
    if ( mode && mode.level > 0 ) {
      itemData.system.preparation.mode = mode["preparation.mode"];
    }
```
Personally, I would add braces here, since the `if` statement has an `else if` statement, but ultimately that's up to @Fyorl and/or @arbron.